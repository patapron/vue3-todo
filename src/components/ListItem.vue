<template>
  <v-list-item-avatar start>
    <v-checkbox
      v-model="props.element.completed"
      hide-details
      @change="changeEmit(props.element)"
    ></v-checkbox>
  </v-list-item-avatar>

  <v-list-item-header>
    <v-list-item-title
      ><v-btn v-if="!editing" variant="flat" @click="editToggle">{{
        props.element.name
      }}</v-btn>
      <v-text-field
        v-else
        label="Include files"
        hide-details
        v-model="props.element.name"
        v-on:keyup.enter="changeEmit(props.element)"
      ></v-text-field>
    </v-list-item-title>
  </v-list-item-header>
  <v-list-item-avatar end>
    <v-btn
      @click="deleteEmit(props.element)"
      variant="text"
      color="grey lighten-1"
      icon="mdi-close"
    ></v-btn>
  </v-list-item-avatar>
</template>

<script>
import { ref, inject } from "vue";

export default {
  name: "ListItem",
  props: { element: null },
  setup(props, context) {
    const editing = ref(false);
    const bus = inject("bus");

    function editToggle() {
      editing.value = !editing.value;
    }

    function changeEmit(event) {
      editToggle();
      bus.emit("delete-event", event);
    }

    function deleteEmit(event) {
      if (!editing.value) {
        bus.emit("delete-event", event);
      }
    }
    return {
      editing,
      editToggle,
      changeEmit,
      deleteEmit,
      props,
    };
  },
};
</script>
