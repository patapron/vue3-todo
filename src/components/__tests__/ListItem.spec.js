/* eslint-disable prettier/prettier */
import ListItem from "@/components/ListItem.vue";
import { shallowMount } from "@vue/test-utils";

// import { MODULE_CODE } from '../../../config/';

describe("List", () => {
  let wrapper;
  let element;

  beforeEach(() => {
    element = { id: 1, name: "test", completed: false };
    wrapper = shallowMount(ListItem, {
      mocks: {
        $t: () => "Test text",
        $can: () => true,
      },
      propsData: {
        element,
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("module name", () => {
    expect(wrapper.vm.props.element.name).toEqual("test");
  });
});
