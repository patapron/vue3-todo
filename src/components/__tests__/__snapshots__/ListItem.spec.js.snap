// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<v-list-item-avatar start="">
  <v-checkbox modelvalue="false" hide-details=""></v-checkbox>
</v-list-item-avatar>
<v-list-item-header>
  <v-list-item-title>
    <v-btn variant="flat">test</v-btn>
  </v-list-item-title>
</v-list-item-header>
<v-list-item-avatar end="">
  <v-btn variant="text" color="grey lighten-1" icon="mdi-close"></v-btn>
</v-list-item-avatar>
`;
